linker/linker_test.go

```diff
@@ -498,6 +498,8 @@ func TestLinkerValidation(t *testing.T) {
 			expectedErr: "foo.proto:1:95: field com.google.Foo.str: unknown type google.protobuf.StringValue; resolved to com.google.protobuf.StringValue which is not defined; consider using a leading dot",
 		},
 		"success_group_in_custom_option": {
+			// Groups must be referred by lower-case field name in option name.
+			// Hence "bar" instead of "Bar" in the option at the end.
```

Comment:
I've added comments to this sequence of test cases, hoping to "tell a tale", to make the gist of the test cases a little easier to follow.